File: examples/06_observables/observables.cpp
// Example 06: Observables — Building quantum observables.

#include "quantum_info/sparse_observable.hpp"
#include <complex>
#include <iostream>
#include <vector>
#include <utility>

using namespace Qiskit::quantum_info;

int main() {
    // Build from list
    std::vector<std::pair<std::string, std::complex<double>>> terms;
    terms.push_back(std::make_pair("IIXY", std::complex<double>(1.0, 0.0)));
    terms.push_back(std::make_pair("ZZII", std::complex<double>(-0.5, 0.0)));
    auto obs1 = SparseObservable::from_list(terms);
    std::cout << "Observable from list:" << std::endl;

    // Identity and zero
    auto identity = SparseObservable::identity(4);
    auto zero = SparseObservable::zero(4);
    std::cout << "Created identity and zero observables" << std::endl;

    // Arithmetic
    auto combined = obs1;
    combined += identity;
    std::cout << "Combined obs1 + identity" << std::endl;

    return 0;
}

File: examples/06_observables/observables.py
"""Example 06: Observables — Building quantum observables."""

from qiskit.quantum_info import SparsePauliOp

# Build from label
obs1 = SparsePauliOp.from_list([("IIXY", 1.0), ("ZZII", -0.5)])
print("Observable from list:")
print(obs1)

# Identity and zero
identity = SparsePauliOp.from_list([("IIII", 1.0)])
zero = SparsePauliOp.from_list([("IIII", 0.0)])
print(f"\nIdentity:\n{identity}")

# Arithmetic
combined = obs1 + identity
print(f"\nobs1 + identity:\n{combined}")
